src/main/java/org/cristalise/lookup/ldap/LDAPLookup.java

```diff
@@ -435,13 +435,16 @@ public ItemPath getItemPath(String uuid) throws ObjectNotFoundException, Invalid
         try {
         	ItemPath item = new ItemPath(uuid);
             LDAPEntry anEntry=mLDAPAuth.getAuthObject().read(getDN(item)+mLocalPath,attr);
-            String type = LDAPLookupUtils.getFirstAttributeValue(anEntry, "objectClass");
-            if (type.equals("cristalentity"))
-                return item;
-            else if (type.equals("cristalagent"))
-                return new AgentPath(item);
-            else
-                throw new ObjectNotFoundException("Not an entity '"+uuid+"'");
+            String[] values = LDAPLookupUtils.getAllAttributeValues(anEntry, "objectclass");
+            if (values != null) {
+            	for (String type : values) {
+            		if (type.equals("cristalentity"))
+                        return item;
```

Comment:
I was waiting for an update to the PR. As I said above, you can't count on a 'cristalentity' being an Item unless you're sure there's no 'cristalagent'.